crates/wasmtime/src/module.rs

```diff
@@ -1042,6 +1042,47 @@ impl Module {
         self.inner.memory_images()?;
         Ok(())
     }
+
+    // XXX: Only for use within this repo's workspace.
+    #[doc(hidden)]
+    pub fn compiled_functions_info(&self) -> CompiledFunctionsInfo<'_> {
+        CompiledFunctionsInfo {
+            text: self.code_object().code_memory().text(),
+            func_locs: self
+                .compiled_module()
+                .finished_functions()
+                .map(|(f, _)| *self.compiled_module().func_loc(f))
+                .collect(),
+            address_map_section: self.code_object().code_memory().address_map_data(),
+        }
+    }
+}
+
+// XXX: Only for use within this repo's workspace.
+#[doc(hidden)]
+pub struct CompiledFunctionsInfo<'a> {
+    text: &'a [u8],
+    func_locs: Vec<FunctionLoc>,
+    address_map_section: &'a [u8],
+}
+
+// XXX: Only for use within this repo's workspace.
+#[doc(hidden)]
+pub use wasmtime_environ::{FilePos, FunctionLoc};
```

Comment:
If possible I think it would be best to keep these private. Could iterators be exposed which peel away these types when exposed to outside users?